### PR TITLE
Adding documentation on the scope of functionality

### LINKED ISSUE #80: Document the scope of functionality
The HTTP add on needs to document the functionality it provides.

### Use-Case

This document will set expectations for new users and developers of the project, and also help guide new feature development, include deciding whether to implement a feature at all.

### Specification

- [ ] A `docs/scope.md` document that outlines the scope of functionality, including clear delineation on where the project starts and stops
- [ ] Link to the new document from the README

### PR DESCRIPTION
**Pull request by arschles**
This PR adds a document that outlines the scope of functionality this project covers, for use in setting expectations and guiding future feature decisions.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

Fixes #80

### CODE REVIEW

**Comment by arschles** [comment]
@tomkerkhove we discussed this in the March 2, 2021 developer standup. This is probably not an exhaustive document currently, just a start.

File: docs/scope.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# The Scope of the KEDA HTTP Add On
+
+Running production HTTP servers in Kubernetes is complicated and involves many pieces of infrastructure. The HTTP Add On (called the "add on" hereafter) aims to autoscale these HTTP servers, but does not aim to extend beyond that scope. Generally, this project only aims to do two things:
+
+1. Autoscale arbitrary HTTP servers based on the volume of traffic incoming to it, including to zero
+2 Route HTTP traffic from a given source to an arbitrary HTTP server, as far as we need to efficiently accomplish (1)
```
Comment (tomkerkhove) [inline docs/scope.md:6]:
```suggestion
2. Route HTTP traffic from a given source to an arbitrary HTTP server, as far as we need to efficiently accomplish (1)
```

File: docs/scope.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+The add on only provides this functionality to workloads that _opt in_ to it. We provide more detail below.
+
+## Autoscaling HTTP
+
+To autoscale HTTP servers, the HTTP Add On needs access to metrics that it can report to KEDA, so that KEDA itself can scale the target HTTP server. The mechanism by which the add on does this is to use an [interceptor](../interceptor) and [external scaler](../scaler). An operator watches for a `HTTPScaledObject` resource and creates these components as necessary.
+
+The HTTP Add On only includes the necessary infrastructure to respond to new, modified, or deleted `HTTPScaledObject`s, and when one is created, the add on only creates the infrastructure needed specifically to accomplish autoscaling.
+
+>As stated above, the current architecture requires an "interceptor," which needs to proxy incoming HTTP requests in order to provide autoscaling metrics. That means the scope of the HTTP add on currently needs to include the app's network traffic routing system.
```
Comment (tomkerkhove) [inline docs/scope.md:16]:
```suggestion
>As stated above, the current architecture requires an "interceptor," which needs to proxy incoming HTTP requests in order to provide autoscaling metrics. That means the scope of the HTTP add on currently needs to include the app's network traffic routing system.

To learn more, we recommend reading about our [design](design.md) or go through our [FAQ](faq.md).

```

**Review by tomkerkhove — APPROVED** [review]

**Comment by arschles** [comment]
@tomkerkhove I made changes according to your comments. can you PTAL again?

File: docs/scope.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
```
Comment (tomkerkhove) [inline docs/scope.md:16]:
```suggestion
>As stated above, the current architecture requires an "interceptor", which needs to proxy incoming HTTP requests in order to provide autoscaling metrics. That means the scope of the HTTP add-on currently needs to include the app's network traffic routing system.
```

**Review by tomkerkhove — APPROVED** [review]

**Pull request merged**